### ISSUE TITLE
`jit(pmap)` on key arrays raises an error in `pmap` lowering

### ISSUE THREAD

**Issue by Joshuaalbert** [google/jax#20392]
### Description

While not really a bug, I was surprised to find a rogue variable `jax.random.key` that seems to be similar to PRNGKey but fails for the case`jit(pmap)`. Copilot dropped this surprise, as an autocomplete that I didn't notice until later on during debugging.

```python
import os

num_devices = 8

os.environ['XLA_FLAGS'] = f'--xla_force_host_platform_device_count={num_devices}'

import jax


def test():
    def single_fn(key):
        return jax.random.normal(key, ())

    def pmap_batched_fn(key):
        batch_size = num_devices
        keys = jax.random.split(key, batch_size)
        pmap_fn = jax.pmap(single_fn)
        return pmap_fn(keys)

    key = jax.random.key(0) # Some rogue jax.random.key
    # key = jax.random.PRNGKey(0) # Works of course

    # Passes pmap(...)
    result_pmap_no_jit = pmap_batched_fn(key)

    # Fails jit(pmap(...))
    result_pmap_jit = jax.jit(pmap_batched_fn)(key)


```

### System info (python version, jaxlib version, accelerator, etc.)

Present at least in JAX versions 0.4.20 to 0.4.25

**Comment by froystig** [comment]
`jax.random.key` is far from rogue. See:
* https://jax.readthedocs.io/en/latest/jax.random.html
* https://jax.readthedocs.io/en/latest/jep/9263-typed-keys.html
* #9263

The function was introduced quietly a few versions ago. We only recently switched the module documentation to mention it (in #17741), so it isn't in the documentation for a release yet.

So, this looks like a bug! Thanks for filing.

Also, since we changed the docs in the current release, we should remember to mention it in the changelog.

fyi @jakevdp
